fog/sample-paykit/src/cached_tx_data/mod.rs

```diff
@@ -40,6 +47,8 @@ const MAX_INPUTS: usize = mc_transaction_core::constants::MAX_INPUTS as usize;
 /// balance data.
 #[derive(Clone)]
 pub struct CachedTxData {
+    /// The account key we are tracking tx data for.
+    account_key: AccountKey,
```

Comment:
One reason I didn't do this is that, the more copies of `AccountKey` we make in memory the harder it is to ensure that we are zeroizing it appropriately. but it does make things simpler. Maybe we should just think of the zeroizing thing as a separate problem to solve and deal with it later.

Reply:
Maybe it should implement the Zeroize stuff?
I understand the desire to ensure it gets zeroized properly but passing it around everywhere feels very janky...

Reply:
yeah i mean, at the level of like, machine code, the ideal thing is that the secret key exists at as few locations as possible and we pass pointers around to it, i think. but having two copies is probably fine.

Reply:
I guess we did fix the zeroizing stuff already, so this is already zeroizing itself:

https://github.com/mobilecoinfoundation/mobilecoin/blob/1528cf64d42e52783fa940f88fc29e47f993fb3e/account-keys/src/account_keys.rs#L192

Reply:
so this is probably not a real concern

Reply:
I think that if we cared that much about the references thing then we probably should've put the keys inside an Arc<>, and that will ensure a single copy.